src/Core/Authoring/PackageBuilder.cs

```diff
@@ -531,6 +534,48 @@ private void AddFiles(string basePath, string source, string destination, string
             Files.AddRange(searchFiles);
         }
 
+        /// <summary>
+        /// Returns true if the path uses a known v3 folder root.
+        /// </summary>
+        private static bool IsKnownV3Folder(string targetPath)
+        {
+            if (targetPath != null)
+            {
+                var parts = targetPath.Split(
+                    new char[] { '\\', '/' },
+                    StringSplitOptions.RemoveEmptyEntries);
+
+                // exclude things in the root of the directory, this is not allowed
+                // for any of the v3 folders.
+                // example: an empty 'native' folder does not have a TxM and cannot be used.
+                if (parts.Length > 1)
+                {
+                    var topLevelDirectory = parts.FirstOrDefault();
+
+                    return KnownFoldersForV3.Any(folder =>

```

Comment:
is there a reason that the reader should pay special attention to this scenario?